control-plane/p4RuntimeSerializer.cpp

```diff
@@ -661,10 +661,11 @@ getMatchFields(const IR::P4Table* table, ReferenceMap* refMap, TypeMap* typeMap)
           typeMap->getType(keyElement->expression->getNode(), true);
         BUG_CHECK(matchFieldType != nullptr,
                   "Couldn't determine type for key element %1%", keyElement);
-
+        while (matchFieldType->is<IR::Type_Newtype>())
```

Comment:
This seems to be just a quick work around to get the P4Info to generate. According to the P4Runtime spec, the type information needs to be preserved for match fields.

Reply:
I think it's a workaround to get around the gtestp4c failure.  I have proposed that the code below is added to minWidthBits() and then the gtestp4c failure is fixed.  This is the code change is included in my PR.  I think this PR should only touch typeChecker.cpp and leave the rest to my PR. 

```
   // Check first if incoming type is bits, otherwise getTypeType will
    // cause failure in getType and p4runtime gtestp4c tests fail.
    if (type->is<IR::Type_Bits>()) {
	  return type->width_bits();
    }
```